### ISSUE TITLE
Chiseled Leaves Fancy & Fast

### ISSUE THREAD

**Issue by kiomadoushi** [ChiselsAndBits/Chisels-and-Bits#251]
I normally play on Fancy graphics, but switched to Fast just to mess around with something. When switching back to Fancy, my chiseled leaves stayed opaque, using Fast graphics. Closing and opening my world didn't do anything either. Whenever I'd change them, they'd turn back to Fancy transparent leaves, but giving them the same shape made them opaque again. Even destroying them and replacing them with a new leaf block and manually chiseling it back into the same shape made them turn into Fast opaque leaves.

I did find one way to fix them: changing them so they're transparent, saving and closing, then reopening. They could then be fixed back into the intended shape without turning opaque again. It seems saving the game saves the graphics state of the leaves, preventing them from changing with graphics options.

EDIT: fix, changing them so they're transparent*, not opaque.

**Comment by AlgorithmX2** [comment]
What version are you using, I was sure that this issue was already fixed.

Minecraft handles fancy / fast graphics in a very stupid way, which requires me to constantly scan the value for changes so I can update my blocks properties appropriately.

**Comment by kiomadoushi** [comment]
12.12 for mc 1.10.2

**Comment by AlgorithmX2** [comment]
This bug was fixed in 13.6 on 1.11 I'm not sure its worth back-porting this, since it only matters if your changing your settings you usually leave untouched and doesn't really cause any significant issues.

**Comment by kiomadoushi** [comment]
![2017-03-24_13 39 35](https://cloud.githubusercontent.com/assets/9398848/24312791/6e5abb5e-1097-11e7-9dcc-7287479efb7b.png)

Maybe not significant, but it does still leave these ugly patches that you then can't easily build over... And is 1.10.2 really considered that much of back-porting? That's odd to hear; almost all of my mods have no 1.11 version.

**Comment by AlgorithmX2** [comment]
Its not that much of a back-port, but it still requires time, and I am sorely lacking in that deparment.

For now i'll leave this open and when I have some time maybe I'll pull it back to release on the 1.10 branch.

For future reference: https://github.com/AlgorithmX2/Chisels-and-Bits/commit/d3d772fbdb9464163fa0f4e6eeb1c6e90757b10d

**Comment by AlgorithmX2** [comment]
In the mean time if you you are flipping back and forth you can use the Shift+F3 ( I think ) to reload the textures which will triggers the update as well.

**Comment by kiomadoushi** [comment]
Thanks, the texture reload actually did work. I still had like 20 leaf blocks to fix, and it was getting annoying reopening everything each time.

**Comment by AlgorithmX2** [comment]
Turns out that this was already backported, and it broke on both 1.10 and 1.11.

**Comment by AlgorithmX2** [comment]
please use https://www.dropbox.com/s/gtqsx2s57jrg5g8/chiselsandbits-12.0.jar?dl=0 to confirm the issue is fixed.

**Comment by kiomadoushi** [comment]
Yes, it works.
That does allow switching fast to fancy, and fancy to fast. No invisible leaves nor opaque leftover.

**Comment by kiomadoushi** [comment]
ALTHOUGH, I think I found a new bug. Should I just make a new issue? It seems to have come up when chiseling fast leaves...

Chiseled oak leaves form two types of bits. I don't know a better way to describe it than knowing they're not the same bits because it won't count the block as complete if the two mix, removing bits by "same material" leaves the other leaf bits, and deleting all bits in your bag by material only deletes the same leaf bits but not the others.

Found it because this one block is apparently incomplete despite being a full block, and so I broke it down into bits and they didn't all erase when I used "same material".
(I'm sorry. Fix one bug and create another, right?)

**Comment by kiomadoushi** [comment]
Pics to show what I'm talking about:

![2017-03-25_15 43 07](https://cloud.githubusercontent.com/assets/9398848/24326718/d424ff28-1171-11e7-8e6d-aa7d14ee44dc.png)
![2017-03-25_15 43 14](https://cloud.githubusercontent.com/assets/9398848/24326719/d4265c60-1171-11e7-8909-6b80263de676.png)

Neat effect, though, the "incomplete" one staying fancy until chiseled...
Edit: Also won't let that action be undone with the Undo button...

**Comment by AlgorithmX2** [comment]
This might be related to https://github.com/AlgorithmX2/Chisels-and-Bits/issues/158

basically leaves contain a number of odd states like if they are natural leaves, or placed leaves.  So if you say, place a leaf, chisel it, then chisel a natural leaf, those are technically two completely different types of leaves.

The only way to fix that is to special case leaves... which probably doesn't end well.

**Comment by kiomadoushi** [comment]
While I would say that makes sense, none of these leaf blocks I was testing on naturally grew there, and all came from the same stack in my inventory. No bits were on me nor in the bag I had on me. All the bits used were originally from the "incomplete" one and the one chiseled into basic stairs to the left of it, both of which were just placed by me right before testing.

In detail, I switched to fast from the menu, opened my world, found a place to throw down those 3 oak leaf blocks, and chiseled middle leaf block into stairs. Switched to fancy, and block changed as expected. So I chiseled right block into a slab. Switched back to fast, and blocks were still looking good. Reported the new .jar fixed it.

Then I rebuilt the right one, and it didn't change from a chiseled block back to a vanilla block. Broke by material, and everything on top only was deleted. Broke the bottom half and went to delete bits, but only half of them deleted. Deleted the other half, placed a new block, broke by material (breaking them all as expected) and then rebuilt it in planes. Some planes didn't build entirely, stopping short a couple lines (probably when it was switching bits, idk the cause). Took screenshot #1, removed by material, then took screenshot #2.

Regardless, the first problem is fixed, so yay.
